src/System.Private.Xml/tests/Misc/XmlUrlResolverTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    public class XmlUriResolverTests
+    {
+        [Fact]
+        public void Resolving_RelativeBase_Throws()
+        {
+            var resolver = new XmlUrlResolver();
+            Assert.Throws<NotSupportedException>(() => resolver.ResolveUri(
+                new Uri(Environment.CurrentDirectory + Path.DirectorySeparatorChar, UriKind.Relative), "test.xml"));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetBaseUriAndPath))]
+        public void Resolving_LocalPath_Ok(Uri baseUri, string path)
+        {
+            var resolver = new XmlUrlResolver();
+            Uri resolvedUri = resolver.ResolveUri(baseUri, path);
+
+            Assert.Equal(Path.GetFullPath(Path.Combine(Environment.CurrentDirectory, path)), resolvedUri.LocalPath);
+            Assert.True(resolvedUri.LocalPath.EndsWith(path.Replace('/', Path.DirectorySeparatorChar)));
+        }
+
+        [Theory]
+        [MemberData(nameof(XmlFileTargets))]
+        public void Resolving_OnlyWithBaseUri_Ok(string basePath)
+        {
+            var baseUri = new Uri(Path.GetFullPath(basePath));
+            var resolver = new XmlUrlResolver();
+            Uri resolvedUri = resolver.ResolveUri(baseUri, string.Empty);
+
+            Assert.Equal(Path.GetFullPath(Path.Combine(basePath)), resolvedUri.LocalPath);
+        }
+
+        public static IEnumerable<object[]> GetBaseUriAndPath()
+        {
+            // Base URI as null is the default for internal Xml operation.
+            var baseUris = new List<Uri> { null };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // The case below does not work on Unix, the '#' ends up treated as a fragment and the path is cut there.
```

Comment:
@rmkerr I just want to confirm that this is really expected in Unix. Thanks.

Reply:
I don't think that behavior should be specific to Unix. If you aren't using local path, I would always expect that behavior. Is this occurring only on Unix?

Reply:
Yes, those two only fail in Unix.

Reply:
It is a single issue but it results in two failures, see below. 

```
     System.Xml.Tests.XmlUriResolverTests.Resolving_LocalPath_Ok(baseUri: file:///home/pc/s/corefx/bin/tests/System.Xml.Misc.Tests/netcoreapp-Linux-Debug-x64/, path: \"/f#/t/ë/t#st.xml\") [FAIL]
        Assert.Equal() Failure
                    ↓ (pos 2)
        Expected: /f#/t/ë/t#st.xml
        Actual:   /f
                    ↑ (pos 2)
        Stack Trace:
           /home/pc/s/corefx/src/System.Private.Xml/tests/Misc/XmlUrlResolverTests.cs(29,0): at System.Xml.Tests.XmlUriResolverTests.Resolving_LocalPath_Ok(Uri baseUri, String path)
     System.Xml.Tests.XmlUriResolverTests.Resolving_LocalPath_Ok(baseUri: file:///home/pc/s/corefx/bin/tests/System.Xml.Misc.Tests/netcoreapp-Linux-Debug-x64/, path: \"/f#/ã/ë/tëst.xml\") [FAIL]
        Assert.Equal() Failure
                    ↓ (pos 2)
        Expected: /f#/ã/ë/tëst.xml
        Actual:   /f
                    ↑ (pos 2)
        Stack Trace:
           /home/pc/s/corefx/src/System.Private.Xml/tests/Misc/XmlUrlResolverTests.cs(29,0): at System.Xml.Tests.XmlUriResolverTests.Resolving_LocalPath_Ok(Uri baseUri, String path)
  Finished:    System.Xml.Misc.Tests
```

Reply:
@rmkerr I will merge this PR as it is and open an issue to track the investigation of this issue.

Reply:
That sounds good to me. Thanks for looking into this!

Reply:
Following up on #28486